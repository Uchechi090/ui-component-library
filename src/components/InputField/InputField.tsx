import React, { ChangeEvent } from "react";

import "./InputField.css";

interface InputProps {
  type: "text" | "number" | "email" | "password";
  label: string;
  value: string | number;
  name: string;
  placeholder: string;
  isValid?: boolean;
  disabled?: boolean;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

const InputField: React.FC<InputProps> = ({
  type,
  label,
  value,
  name,
  placeholder,
  isValid,
  disabled,
  onChange
}) => {
  return (
    <div className="input-wrapper" data-testid="input-wrapper">
      <label htmlFor={label}>{label}</label>
      <input
        data-testid="input-field"
        type={type}
        id={label}
        value={value}
        name={name}
        placeholder={placeholder}
        onChange={onChange}
        disabled={disabled}
      />
      {!isValid && <p data-testid="error-validation" className="error">Please check that your input is valid</p>}
    </div>
  );
};

export default InputField;
