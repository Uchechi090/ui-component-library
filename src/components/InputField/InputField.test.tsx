import React from "react";
import { render, fireEvent } from "@testing-library/react";

import InputField from "./InputField";

describe("<InputField />", () => {
  const mockChangeValue = jest.fn();

  it("should show props", () => {
    const { getByTestId, asFragment } = render(
      <InputField
        label={"Name"}
        onChange={mockChangeValue}
        type={"text"}
        value={"Jane Doe"}
        name={"name"}
        placeholder={"Please enter your name"}
        isValid={true}
      />
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByTestId("input-field").getAttribute("value")).toBe("Jane Doe");
    expect(getByTestId("input-field").getAttribute("type")).toBe("text");
    expect(getByTestId("input-field").getAttribute("name")).toBe("name");
    expect(getByTestId("input-field").getAttribute("placeholder")).toBe(
      "Please enter your name"
    );

    fireEvent.change(getByTestId("input-field"), { target: { value: "Jane" } });
    expect(mockChangeValue).toBeCalled();
  });

  it("should test for when isValid is false", () => {
    const { getByTestId, asFragment } = render(
      <InputField
        label={"Name"}
        onChange={mockChangeValue}
        type={"text"}
        value={"Jane Doe"}
        name={"name"}
        placeholder={"Please enter your name"}
        isValid={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("error-validation")).toHaveTextContent(
      /Please check that your input is valid/
    );
  });

  it("should test for disabled", () => {
    const { getByTestId, asFragment } = render(
      <InputField
        label={"Name"}
        onChange={mockChangeValue}
        type={"text"}
        value={"Jane Doe"}
        name={"name"}
        placeholder={"Please enter your name"}
        isValid={true}
        disabled
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("input-field")).toHaveAttribute("disabled");
  });
});
