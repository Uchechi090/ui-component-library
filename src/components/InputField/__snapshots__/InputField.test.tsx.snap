// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> should show props 1`] = `
<DocumentFragment>
  <div
    class="input-wrapper"
    data-testid="input-wrapper"
  >
    <label
      for="Name"
    >
      Name
    </label>
    <input
      data-testid="input-field"
      id="Name"
      name="name"
      placeholder="Please enter your name"
      type="text"
      value="Jane Doe"
    />
  </div>
</DocumentFragment>
`;

exports[`<InputField /> should test for disabled 1`] = `
<DocumentFragment>
  <div
    class="input-wrapper"
    data-testid="input-wrapper"
  >
    <label
      for="Name"
    >
      Name
    </label>
    <input
      data-testid="input-field"
      disabled=""
      id="Name"
      name="name"
      placeholder="Please enter your name"
      type="text"
      value="Jane Doe"
    />
  </div>
</DocumentFragment>
`;

exports[`<InputField /> should test for when isValid is false 1`] = `
<DocumentFragment>
  <div
    class="input-wrapper"
    data-testid="input-wrapper"
  >
    <label
      for="Name"
    >
      Name
    </label>
    <input
      data-testid="input-field"
      id="Name"
      name="name"
      placeholder="Please enter your name"
      type="text"
      value="Jane Doe"
    />
    <p
      class="error"
      data-testid="error-validation"
    >
      Please check that your input is valid
    </p>
  </div>
</DocumentFragment>
`;
